src/wallet/wallet2.cpp

```diff
@@ -8640,28 +8640,42 @@ uint64_t wallet2::import_key_images(const std::vector<std::pair<crypto::key_imag
 		m_transfers[n].m_key_image_partial = false;
 	}
 
-	if(check_spent)
+	if(!check_spent)
```

Comment:
`spend` and `unspend` from https://github.com/ryo-currency/ryo-currency/pull/141/files?w=1#diff-9d580668dab930045943902048e2ac22R8661 should be set to zero before this early exit.